core/domain/topic_services.py

```diff
@@ -247,23 +250,55 @@ def save_new_topic(committer_id, topic):
 
 
 def apply_change_list(topic_id, change_list):
-    """Applies a changelist to a topic and returns the result.
+    """Applies a changelist to a topic and returns the result. The incoming
+    changelist should not have simultaneuous creations and deletion of
+    subtopics.
 
     Args:
         topic_id: str. ID of the given topic.
         change_list: list(TopicChange). A change list to be applied to the given
             topic.
 
+    Raises:
+        Exception. The incoming changelist had simultaneuous creation and
+            deletion of subtopics.
     Returns:
-        Topic. The resulting topic domain object.
+        dict, list(int), list(int). A dict with two fields: topic and
```

Comment:
Btw, I think I commented on this before, but don't return a dict. Return topic and modified_subtopic_pages separately.

Reply:
Done.